src/Concerns/TransactionManager.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Jenssegers\Mongodb\Concerns;
+
+use Closure;
+use Exception;
+use MongoDB\Driver\Session;
+
+use function MongoDB\with_transaction;
+
+trait TransactionManager
+{
+    /**
+     * Use the existing or create new session and start a transaction in session
+     *
+     * In version 4.0, MongoDB supports multi-document transactions on replica sets.
+     * In version 4.2, MongoDB introduces distributed transactions, which adds support for multi-document transactions on sharded clusters and incorporates the existing support for multi-document transactions on replica sets.
+     * To use transactions on MongoDB 4.2 deployments(replica sets and sharded clusters), clients must use MongoDB drivers updated for MongoDB 4.2.
+     *
+     * @see https://docs.mongodb.com/manual/core/transactions/
+     * @return void
+     */
+    public function beginTransaction(array $options = [])
+    {
+        $session = $this->getSession();
+
+        if (!$session) {
+            $session = $this->connection->startSession();
+            $this->session = $session;
+        }
+
+        $session->startTransaction($options);
+    }
+
+    /**
+     * commit transaction in this session and close this session
+     * @return void
+     */
+    public function commit()
+    {
+        if ($session = $this->getSession()) {
+            $session->commitTransaction();
+        }
+    }
+
+    /**
+     * rollback transaction in this session and close this session
+     * @return void
+     */
+    public function rollBack($toLevel = null)
+    {
+        if ($session = $this->getSession()) {
+            $session->abortTransaction();
+        }
+    }
+
+    /**
+     * get now session if it has session
+     * @return Session|null
+     */
+    public function getSession(): ?Session
+    {
+        return $this->session ?? null;
```

Comment:
in case if we tried to set wrong value PHP will return **TypeError**. 
`TypeError : Cannot assign int to property Jenssegers\Mongodb\Connection::$session of type ?MongoDB\Driver\Session `

Reply:
I previously commented on the property declaration in `Connection.php` in https://github.com/levon80999/laravel-mongodb/pull/3/files#r993071756. This PR doesn't show the property as having a type hint, so what is responsible for the TypeError?

Reply:
So now I moved `session` property declaration in `TransactionManager.php` and set `?Session` type for it. And also set return type of `getSession()` method `?Session`. As this trait is used in `Connection.php` in case if we tried to set wrong value, we will get this error `TypeError : Cannot assign int to property Jenssegers\Mongodb\Connection::$session of type ?MongoDB\Driver\Session`